test/acceptance/helpers/snapshot.js

```diff
@@ -0,0 +1,81 @@
+import http from 'http'
+import puppeteer from 'puppeteer'
+
+/**
+ * Similar to our Puppeteer or PercyScript SDKs - uses a Puppeteer page to
+ * inject agent client code and capture the DOM, and sends that DOM with `name`
+ * and `options` to agent's running server.
+ *
+ * Does not use functions defined in sdk-utils to avoid compiling and to allow
+ * errors to bubble up through our tests. Returns the captured DOM string for
+ * assertions within tests.
+ *
+ * @param {object} page - Puppeteer page objectg
+ * @param {string} name - Snapshot name
+ * @param {object} options - Snapshot options
+ * @returns {string} The captured DOM string
+ */
+async function snapshot(page, name, options = {}) {
+  await page.addScriptTag({
+    url: 'http://localhost:5338/percy-agent.js'
```

Comment:
I quite like this. We could easily remove the SDK util in favor of this for other SDKs

Reply:
The possible caveat is CSP issues. I had to bypass it to inject this script into the dummy express app.